search/lib/api/provider.js

```diff
@@ -4,38 +4,42 @@ const { assertValid, schemas } = require('../validator');
 const settings = require('../settings');
 const cmr = require('../cmr');
 
-function convertProvider (event, provider) {
-  const providerId = provider['provider-id'];
-  return {
-    id: providerId,
-    title: provider['short-name'],
-    description: `Root catalog for ${providerId}`,
-    stac_version: settings.stac.version,
-    links: [
-      wfs.createLink('self', generateAppUrl(event, `/${providerId}`),
-        'Root endpoint for this provider'),
-      wfs.createLink('root', generateAppUrl(event, '/'),
-        'CMR-STAC Root'),
-      wfs.createLink('collections', generateAppUrl(event, `/${providerId}/collections`),
-        'Collections for this provider'),
-      wfs.createLink('search', generateAppUrl(event, `/${providerId}/search`),
-        'STAC Search endpoint for this provider')
-    ]
-  };
-}
-
 async function getProvider (request, response) {
   try {
     const providerId = request.params.providerId;
     logger.info(`GET /${providerId}`);
     const event = request.apiGateway.event;
+
+    // validate that providerId is valid
     const providerList = await cmr.getProviders();
     const isProvider = providerList.filter(providerObj => providerObj['provider-id'] === providerId);
     if (isProvider.length === 0) throw new Error(`Provider [${providerId}] not found`);
-    const provider = convertProvider(event, {
-      'provider-id': providerId,
-      'short-name': providerId
+
+    const links = [
+      wfs.createLink('self', generateAppUrl(event, `/${providerId}`),
+        'Provider catalog'),
+      wfs.createLink('root', generateAppUrl(event, '/'),
+        'CMR-STAC Root catalog'),
+      wfs.createLink('collections', generateAppUrl(event, `/${providerId}/collections`),
+        'Provider Collections'),
+      wfs.createLink('search', generateAppUrl(event, `/${providerId}/search`),
+        'Provider Item Search')
+    ];
+
+    const childLinks = (await cmr.getProvider(providerId)).map((collection) => {
+      return wfs.createLink(
+        'child',
+        generateAppUrl(event, `/${providerId}/collections/${collection['concept-id']}`),
+        collection['entry-title']);
     });
+
+    const provider = {
+      id: providerId,
+      title: providerId,
+      description: `Root catalog for ${providerId}`,
+      stac_version: settings.stac.version,
+      links: [...links, ...childLinks]
+    };
     await assertValid(schemas.catalog, provider);
     response.status(200).json(provider);
```

Comment:
`.status(200)` is redundant when using `.json` unless you need to set a different one